spring-cloud-gateway-server-mvc/src/test/java/org/springframework/cloud/gateway/server/mvc/ServerMvcIntegrationTests.java

```diff
@@ -1080,9 +1096,9 @@ public RouterFunction<ServerResponse> gatewayRouterFunctionsSetPathPost() {
 		public RouterFunction<ServerResponse> gatewayRouterFunctionsStripPrefix() {
 			// @formatter:off
 			return route(GET("/long/path/to/get"), http())
-					.filter(new HttpbinUriResolver())
 					.filter(stripPrefix(3))
 					.filter(addRequestHeader("X-Test", "stripPrefix"))
+					.filter(new HttpbinUriResolver(true))
```

Comment:
Help me understand why this is needed?

Reply:
Honestly, I can't quite remember the details of why I had to do this as it's been a while since I worked on this. I was struggling getting tests to pass, despite believing that they should. The `HttpbinUriResolver` in its current state (without my changes) seems to return a new URI, without any path information being preserved:

```java
return URI.create(String.format("http://%s:%d", host, port));
```

and since that was earlier in the filter chain than `stripPrefix` etc., that seemed to result in the path being chopped off before the `stripPrefix` filter was executed.

For the `stripPrefixPostWorks` test - If I undo all of my changes to `HttpbinUriResolver`, and put it back to the original position in the filter chain, then the HttpBin container seems to respond with a 500 internal server error, but with my code in place, it returns 200. Further, if I leave my modifications in place and just change the position in this test to move it further down the chain, I get the same outcome (500 response), but the requests sent to HttpBin appear to be identical to the one where it returns a 200.

I may be going completely down the wrong path here, but I'm convinced that either I'm doing something wrong or something is not quite right.

I'd appreciate any pointers for getting these two tests correct